File: springapp/src/main/java/com/examly/springapp/service/BookingService.java
package com.examly.springapp.service;

import com.examly.springapp.exception.BookingNotFoundException;
import com.examly.springapp.model.Booking;
import com.examly.springapp.repository.BookingRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class BookingService {

    @Autowired
    private BookingRepository bookingRepository;

    public Booking addBooking(Booking booking) {
        // Simple fare calculation mock
        booking.setFare(200 + Math.random() * 100);
        if (booking.getDriverName() == null) {
            booking.setDriverName("Default Driver");
            booking.setBikeImage("bike.png");
        }
        return bookingRepository.save(booking);
    }

    public List<Booking> getAllBookings() {
        return bookingRepository.findAll();
    }

    public Booking getBookingById(Long id) {
    return bookingRepository.findById(id)
            .orElseThrow(() -> new BookingNotFoundException("Booking not found with ID: " + id));
}

    public Booking updateBooking(Long id, Booking updatedBooking) {
        Booking existing = getBookingById(id);
        existing.setCustomerName(updatedBooking.getCustomerName());
        existing.setPickupLocation(updatedBooking.getPickupLocation());
        existing.setDropLocation(updatedBooking.getDropLocation());
        existing.setRideTime(updatedBooking.getRideTime());
        existing.setFare(updatedBooking.getFare());
        existing.setDriverName(updatedBooking.getDriverName());
        existing.setBikeImage(updatedBooking.getBikeImage());
        return bookingRepository.save(existing);
    }

    public void deleteBooking(Long id) {
        Booking booking = getBookingById(id);
        bookingRepository.delete(booking);
    }
}

File: reactapp/src/services/bookingService.js
import axios from "axios";

const API_URL = "https://8080-fddecedccde329052728bccfaccecftwo.premiumproject.examly.io/api/bookings";

class BookingService {
  getAllBookings() {
    return axios.get(API_URL);
  }

  getBookingById(id) {
    return axios.get(`${API_URL}/${id}`);
  }

  addBooking(booking) {
    return axios.post(API_URL, booking);
  }

  updateBooking(id, booking) {
    return axios.put(`${API_URL}/${id}`, booking);
  }

  deleteBooking(id) {
    return axios.delete(`${API_URL}/${id}`);
  }
}

export default new BookingService();
